File: src/main/java/com/project/pantau/dto/report/NearbyReportResponse.java
package com.project.pantau.dto.report;

import com.project.pantau.dto.category.CategoryResponse;
import com.project.pantau.enums.ReportStatus;
import lombok.Builder;

import java.time.LocalDateTime;
import java.util.UUID;

@Builder
public record NearbyReportResponse(
        UUID id,
        CategoryResponse category,
        String photoUrl,
        ReportStatus status,
        double latitude,
        double longitude,
        LocalDateTime createdAt
) {
}

File: internal/dto/report/nearby_report_response.go
package report

import (
	"github.com/google/uuid"
	"pantau/internal/dto/category"
	"pantau/internal/entity"
	"time"
)

type NearbyReportResponse struct {
	ID          uuid.UUID                 `json:"id"`
	Category    category.CategoryResponse `json:"category"`
	Description *string                   `json:"description"`
	PhotoURLs   []string                  `json:"photo_urls"`
	Latitude    float64                   `json:"latitude"`
	Longitude   float64                   `json:"longitude"`
	Status      entity.ReportStatus       `json:"status"`
	CreatedAt   time.Time                 `json:"created_at"`
}
